llm-vertex.el

```diff
@@ -176,8 +176,13 @@ the key must be regenerated every hour."
   (llm-provider-extract-function-calls provider (json-read-from-string response)))
 
 (cl-defmethod llm-provider-chat-request ((_ llm-google) prompt _)
-  (llm-provider-utils-combine-to-user-prompt prompt llm-vertex-example-prelude)
+  (llm-provider-utils-combine-to-system-prompt prompt llm-vertex-example-prelude)
   (append
+   (when-let ((first (car (llm-chat-prompt-interactions prompt))))
```

Comment:
The `when-let` is confusing me.  When would there not be a first interaction?

Reply:
I vaguely remember copying this from somewhere in the original version of the code. You're right that it is not needed, so this seemed a good time to change it just to a let.